### DISCUSSION TITLE
Target latest APIs and modernize implementation

### DISCUSSION THREAD

**Discussion by seankhl** [FossifyOrg/General-Discussion#35] [Ideas]
I'm not suggesting shifting away from the base design and simplicity that these apps have always had -- that's something that differentiates them and makes them a unique proposition.

But I do think they can be further modernized while retaining these traits, which means moving toward more complete usage of AndroidX, Jetpack Compose, Material 3, defaulting to Material You utilization (with still the option to set custom colors), etc.

Any privacy or security considerations impacted by supporting older versions of android should result in dropping the old APIs and moving toward newer ones, e.g. with permissions APIs. Any code base simplification that can happen in light of this can then be done as well.

**Comment by inson1** [comment]
What I know. SMT apps has been slowly rewritting into Jetpack compose. But its not yet done. And I dont think it should be the priority. - Bugs should be the priority

**Reply by naveensingh** [reply]
That's barely done, it'll take a lot of time before we can fully adopt compose, and yes, it won't be a priority at the moment. But any new features or screens should use Jetpack Compose.

**Comment by inson1** [comment]
Are there really any privacy and security bad impacts by using older APIs?

**Comment by naveensingh** [comment]
@seankhl Of course. We'll slowly get there. We'll focus on security and reliability first, everything else comes after that.

Since this is a community-supported project, we won't be doing needless rewrites just for the sake of rewriting code without any real benefits to the end user.
